### DISCUSSION TITLE
Sibling queries only match the first node instead of the whole section?

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#1341] [Q&A] [answered]
Using tree-sitter-cli 0.20.0. Consider the following TLA+ file:
```tla
---- MODULE Test ----
def1 == 1
---------------------
def2 == 2
---------------------
def3 == 3
---------------------
def4 == 4
=====================
```
Generating the following parse tree:
```
(source_file (module (single_line) (identifier) (single_line)
  (unit (operator_definition (name) (def_eq) (number)))
  (unit (single_line))
  (unit (operator_definition (name) (def_eq) (number)))
  (unit (single_line))
  (unit (operator_definition (name) (def_eq) (number)))
  (unit (single_line))
  (unit (operator_definition (name) (def_eq) (number)))
(double_line)))
```
I want to write a query that matches each section between `(unit (single_line))` nodes, for folding purposes. I try:
```
(
  (unit (single_line))
  .
  (unit (operator_definition))+
  .
  (unit (single_line))
) @section
```
However, the output of `tree-sitter query` is:
> tree-sitter query .\queries\test.scm .\Test.tla
.\Test.tla
  pattern: 0
    capture: section, start: (2, 0), text: "---------------------"
  pattern: 0
    capture: section, start: (4, 0), text: "---------------------"

The capture only matches the very first node in the sibling query, not any of the subsequent nodes. This is how I would expect the query to behave if it were instead as follows:
```
(
  (unit (single_line)) @section
  .
  (unit (operator_definition))+
  .
  (unit (single_line))
)
```
and indeed it then produces the exact same set of captures. Is this intended behavior?

Bonus question: is there any way to negate matches of nodes, in the same way you can negate matches of fields?

**Comment by maxbrunsfeld** [comment, marked as answer]
Yeah, captures need to go on individual nodes, not parenthesized groups of nodes. So you’d just want to put a capture name after each ‘unit’ node. You can reuse the same name if you’re not trying to differentiate them.
